vtex/utils/legacy.ts

```diff
@@ -32,13 +32,24 @@ const PAGE_TYPE_TO_MAP_PARAM = {
 
 const segmentsFromTerm = (term: string) => term.split("/").filter(Boolean);
 
+const DEFAULT_SEGMENT_RATE_LIMIT = 10;
+
 export const pageTypesFromPathname = async (
   term: string,
   ctx: AppContext,
 ) => {
-  const segments = segmentsFromTerm(term);
+  let segments = segmentsFromTerm(term);
   const { vcsDeprecated } = ctx;
 
+  const limit = ctx.segmentRateLimit || DEFAULT_SEGMENT_RATE_LIMIT;
+
+  if (segments.length > limit) {
```

Comment:
Shouldn't you splice this above? Maybe in the beginning of the listing page?

This might introduce inconsistency results (e.g: getting product data one way and page type another)

Reply:
I don't know, I think we can expected inconsistency behaviours after limit. Maybe return null?

Slicing it from pathname in the beggining of loader looks like too complex, working with map parameter for example.